apps/src/storage/levelbuilder/Dataset.jsx

```diff
@@ -14,17 +37,58 @@ class Dataset extends React.Component {
     // it as an array."
     // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
     tableRecords: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
-      .isRequired
+      .isRequired,
+    onUploadComplete: PropTypes.func.isRequired
+  };
+
+  state = {
+    notice: null,
+    isError: false
+  };
+
+  displayNotice = (notice, isError) => {
+    this.setState({notice, isError}, () =>
+      setTimeout(() => this.setState({notice: null, isError: false}), 5000)
+    );
+    window.scrollTo(0, 0);
   };
+
+  importCsv = (csv, onComplete) => {
+    $.ajax({
+      url: `/datasets/${this.props.tableName}`,
+      method: 'PUT',
+      contentType: 'application/json',
+      data: JSON.stringify({csv_data: csv})
+    })
+      .done(data => {
+        this.props.onUploadComplete(
+          this.props.tableName,
+          Object.values(data.records),
+          data.columns
+        );
+        this.displayNotice('Upload succeeeded.', false);
+        onComplete();
+      })
+      .fail(() => this.displayNotice('Upload failed.', false));
+  };
+
   render() {
     return (
       <div>
+        {this.state.notice && (
+          <p style={this.state.isError ? styles.error : styles.success}>
```

Comment:
It seems like there are 3 states here: ShowSuccess, ShowError, and ShowNone. Because of that, it seems like it would be clearer to use an enum to decide what, if anything, is displayed. Other parts of the component could set the enum, and then here, the component could read the enum and decide what string to display and with which formatting.

Reply:
Hmm, I think I disagree. I think of this as the combination of two binary states. First, is there a notice to show (yes/no)? Second, is the notice a success or an error?

Reply:
👍